GoMeet/src/test/java/com/google/sps/UpdateLocationServletTest.java

```diff
@@ -46,48 +56,78 @@
 
   private HttpServletRequest request;
   private HttpServletResponse response;
+  private LocationDao mockedLocationDao;
   
   private final LocalServiceTestHelper helper =
       new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
 
+  private final Gson gson = new Gson();
+
   @Before
   public void setUp() {
     helper.setUp();
     request = mock(HttpServletRequest.class);       
-    response = mock(HttpServletResponse.class);  
+    response = mock(HttpServletResponse.class);
+    mockedLocationDao = mock(LocationDao.class); 
   }
 
   @After
   public void tearDown() {
     helper.tearDown();
   }
 
-  /** Tests if the entity's voteCount is updated. */
+  /** Tests if the keystring from the request is sent to the LocationDAO. */
   @Test
-  public void doPostTest() {
-    // Set up the database
-    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
-    Key entityKey = KeyFactory.createKey("Location", LOCATION_A.getTitle());
-    Entity location = new Entity("Location");
-
-    location.setProperty("title", LOCATION_A.getTitle());
-    location.setProperty("lat", LOCATION_A.getLat());
-    location.setProperty("lng", LOCATION_A.getLng());
-    location.setProperty("note", LOCATION_A.getNote());
-    location.setProperty("voteCount", LOCATION_A.getVoteCount());
-    ds.put(location);
+  public void doPostTest() throws IOException {
+    // Set up request mock
+    String keyString =
+        KeyFactory.keyToString(KeyFactory.createKey("Location", LOCATION_A.getTitle()));
+    when(request.getParameter("key")).thenReturn(keyString);
 
-    String keyString = KeyFactory.keyToString(location.getKey());
+    UpdateLocationServlet servlet = new UpdateLocationServlet();
+    servlet.setDao(mockedLocationDao);
+    servlet.doPost(request, response);
+    
+    try {
+      verify(mockedLocationDao, times(1)).updateVote(keyString);
+      verify(response, times(1)).setStatus(HttpServletResponse.SC_OK);
+    } catch (EntityNotFoundException e) {
```

Comment:
Why are you wrapping this in try/catch?

Reply:
also, if try/catch is necessary, it's probably not necessary to only catch EntityNotFoundException s.

Reply:
Because updateVote() throws an EntityNotFoundException. Would it be better to throw the exception up?

Reply:
It seems like the test is meant to fail if an exception is thrown because in the catch(){} block, there is a single fail() call.

It seems like only catching EntityNotFoundException makes the test somewhat fragile because if any other exception is thrown, the test could pass, but it seems to me like you want the test to fail if there is any failure in any of the calls, since this is a test of a "working" code path.

So, I'm wondering why you wrap the important parts of the test in a try/catch call instead of letting the (or any) exception bubble up into the test framework which I suppose should cause the test to fail anyway.